job/job.go

```diff
@@ -34,6 +43,38 @@ type Job struct {
 	instance *packet.Instance
 }
 
+type connect struct{}
+
+type endpointClient interface {
+	getWorkflowsFromTink(hwID string) (tw.WorkflowContextList, error)
+}
+
+func (co *connect) getWorkflowsFromTink(hwID string) (tw.WorkflowContextList, error) {
+	return client.GetWorkflowsFromTink(hwID)
+}
+
+// isHardwareHavingActiveWorkflow fetches workflows for a given hwID and returns
+// the status true if there is a pending (active) workflow for hwID
+// hwID is the hardware/worker ID corresponding to the MAC
+func isHardwareHavingActiveWorkflow(hwID string) (bool, error) {
+
+	joblog.With("hardwareID", hwID).Info("fetching workflows for hardware")
+
+	wcl, err := connectTo.getWorkflowsFromTink(hwID)
+	if err != nil {
+		return false, err
+	}
+
+	for _, wf := range wcl.WorkflowContexts {
+		if wf.CurrentActionState == 0 {
+			joblog.With("hardwareID", hwID, "workflowID", wf.WorkflowId).Info("found" +
+				" active workflow for hardware")
```

Comment:
just on one line please

Reply:
shouldn't we  be following 80 character line width rule ?

Reply:
done